conda_lock/conda_lock.py

```diff
@@ -23,42 +31,130 @@
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
 
 
-def solve_specs_for_arch(channels: List[str], specs: List[str], platform: str) -> str:
+def ensure_conda(conda_executable: Optional[str]):
+    if conda_executable:
+        if pathlib.Path(conda_executable).exists():
+            return conda_executable
+
+    conda_executable = shutil.which("conda")
+    if conda_executable:
+        return conda_executable
+
+    conda_executable = shutil.which("conda.exe")
+    if conda_executable:
+        return conda_executable
+
+    logging.info(
+        "No existing conda installation found.  Installing the standalone conda solver"
+    )
+    return install_conda_exe()
```

Comment:
Nice to check if conda exists but do we really want to install it for the use?

Reply:
Since we're just placing it in the installed path of our package, we're not so much installing as downloading optional package data?

Reply:
OK.